packages/react-reconciler/src/ReactFiberEvents.js

```diff
@@ -0,0 +1,84 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+import type {Fiber} from './ReactFiber';
+
+import {
+  HostComponent,
+  HostText,
+  HostPortal,
+  SuspenseComponent,
+  Fragment,
+} from 'shared/ReactWorkTags';
+
+export function isFiberSuspenseAndTimedOut(fiber: Fiber): boolean {
+  return fiber.tag === SuspenseComponent && fiber.memoizedState !== null;
+}
+
+export function getSuspenseFallbackChild(fiber: Fiber): Fiber | null {
+  return ((((fiber.child: any): Fiber).sibling: any): Fiber).child;
+}
+
+export function isFiberSuspenseTimedOutChild(fiber: Fiber | null): boolean {
+  if (fiber === null) {
+    return false;
+  }
+  const parent = fiber.return;
+  if (parent !== null && parent.tag === Fragment) {
+    const grandParent = parent.return;
+
+    if (
+      grandParent !== null &&
+      grandParent.tag === SuspenseComponent &&
+      grandParent.stateNode !== null
+    ) {
+      return true;
+    }
+  }
+  return false;
+}
+
+export function getSuspenseFiberFromTimedOutChild(fiber: Fiber): Fiber {
+  return ((((fiber.return: any): Fiber).return: any): Fiber);
+}
+
+export function getEventComponentHostChildrenCount(
+  eventComponentFiber: Fiber,
+): number {
+  let hostChildrenCount = 0;
+
```

Comment:
All of these are DEV-only. Can we make it explicit by wrapping their content in `__DEV__`? And maybe `InDEV` suffixes. So that we don't accidentally try to use them elsewhere.